pkg/image/api/validation/validation.go

```diff
@@ -67,9 +67,24 @@ func validateImage(image *api.Image, fldPath *field.Path) field.ErrorList {
 	return result
 }
 
+func ValidateImageUpdate(newImage, oldImage *api.Image) field.ErrorList {
+	result := validation.ValidateObjectMetaUpdate(&newImage.ObjectMeta, &oldImage.ObjectMeta, field.NewPath("metadata"))
+	result = append(result, ValidateImage(newImage)...)
+
+	return result
+}
+
+// ValidateImageSignature ensures that given signatures is valid.
+func ValidateImageSignature(signature *api.ImageSignature) field.ErrorList {
+	return validateImageSignature(signature, nil)
+}
+
 func validateImageSignature(signature *api.ImageSignature, fldPath *field.Path) field.ErrorList {
-	allErrs := field.ErrorList{}
+	allErrs := validation.ValidateObjectMeta(&signature.ObjectMeta, false, oapi.MinimalNameRequirements, fldPath.Child("metadata"))
```

Comment:
Let's start by disallowing labels and annotations.  I can think of use-cases for them, but I think it will complicate things unnecessarily to start.